app/components/post_draft/draft_input/voice_input/sound_wave.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {random} from 'lodash';
+import React, {useEffect} from 'react';
+import {View} from 'react-native';
+import Animated, {
+    Extrapolation,
+    interpolate,
+    useAnimatedStyle,
+    useSharedValue,
+    withRepeat,
+    withSpring,
+} from 'react-native-reanimated';
+
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+
+const WAVEFORM_HEIGHT = 40;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            height: WAVEFORM_HEIGHT,
+            width: 165,
+            flexDirection: 'row',
+            overflow: 'hidden',
+            justifyContent: 'center',
+            alignItems: 'center',
+        },
+        singleBar: {
+            height: WAVEFORM_HEIGHT,
+            width: 2,
+            backgroundColor: theme.buttonBg,
+            marginRight: 1,
+        },
+    };
+});
+
+// if not playing, height = 0
+// if paused, height = where they are at
+// if playing, then modulate the height
+const SoundWave = () => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const animatedValue = useSharedValue(5);
+
+    const animatedStyles = useAnimatedStyle(() => {
+        const newHeight = interpolate(
+            animatedValue.value,
+            [5, 40],
+            [0, 40],
+            Extrapolation.EXTEND,
+        );
+        return {
+            height: newHeight,
+        };
+    }, [animatedValue.value]);
```

Comment:
This dependency should not be needed. If the value changes, this will be recalculated (since that is how the animated styles work) but you don't need to regenerate the function (that is what the dependencies are for).

Reply:
Corrected